### PR TITLE
CI: Introduce COMMON_MAVEN_ARGS with --fail-at-end

### PR DESCRIPTION
**Pull request by famod**
As discussed here: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/CI.3A.20don't.20abort.20on.20first.20test.20failure

### CODE REVIEW

File: .github/workflows/ci-actions-incremental.yml [hunk 1]
```diff
@@ -33,9 +33,10 @@ on:
 env:
   # Workaround testsuite locale issue
   LANG: en_US.UTF-8
-  NATIVE_TEST_MAVEN_OPTS: "-B --settings .github/mvn-settings.xml --fail-at-end -Dquarkus.native.container-build=true -Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
-  WINDOWS_NATIVE_TEST_MAVEN_OPTS: "-B --settings .github/mvn-settings.xml --fail-at-end -Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
-  JVM_TEST_MAVEN_OPTS: "-e -B --settings .github/mvn-settings.xml -Dtest-containers -Dstart-containers -Dformat.skip -DskipDocs"
+  COMMON_MAVEN_ARGS: "-e -B --settings .github/mvn-settings.xml --fail-at-end"
+  NATIVE_TEST_MAVEN_OPTS: "-Dquarkus.native.container-build=true -Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
+  WINDOWS_NATIVE_TEST_MAVEN_OPTS: "-Dtest-containers -Dstart-containers -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
```
Comment (gsmet) [inline .github/workflows/ci-actions-incremental.yml:38]: Not the best moment to do this :). I changed these same lines in another PR.

Comment (famod) [reply .github/workflows/ci-actions-incremental.yml:38]: Yeah, but should be easy to merge though.

Comment (gsmet) [reply .github/workflows/ci-actions-incremental.yml:38]: This will need a rebase.

Comment (famod) [reply .github/workflows/ci-actions-incremental.yml]: Done.

**Review by gsmet — APPROVED** [review]

**Comment by famod** [comment]
Those Kafka failures are known flakes: #16371

Same for the aborted native messaging tests job: #16415
(Seems we have to set the timeout also for those jobs + probably for failsafe-plugin. Will check in the next days.)

TCKs failure looks very unrelated:
```
2021-04-11T17:42:31.7375043Z 2021-04-11 17:42:31,710 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 0.670s. Listening on: http://localhost:8081
2021-04-11T17:42:31.7376011Z 2021-04-11 17:42:31,711 INFO  [io.quarkus] (main) Profile test activated. 
2021-04-11T17:42:31.7378076Z 2021-04-11 17:42:31,711 INFO  [io.quarkus] (main) Installed features: [cdi, mutiny, smallrye-context-propagation, smallrye-fault-tolerance, smallrye-metrics]
2021-04-11T17:42:36.2182190Z 2021-04-11 17:42:36,068 INFO  [io.quarkus] (main) Quarkus stopped in 0.023s
2021-04-11T17:42:36.2383768Z [ERROR] Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 9.331 s <<< FAILURE! - in org.eclipse.microprofile.fault.tolerance.tck.CircuitBreakerRetryTest
2021-04-11T17:42:36.2392955Z [ERROR] org.eclipse.microprofile.fault.tolerance.tck.CircuitBreakerRetryTest.testCircuitOpenWithMultiTimeoutsAsync  Time elapsed: 0.518 s  <<< FAILURE!
2021-04-11T17:42:36.2397567Z java.lang.AssertionError: serviceC should retry in testCircuitOpenWithMultiTimeouts on iteration 3
2021-04-11T17:42:36.2399147Z 	at org.testng.Assert.fail(Assert.java:96)
2021-04-11T17:42:36.2403940Z 	at org.eclipse.microprofile.fault.tolerance.tck.CircuitBreakerRetryTest.testCircuitOpenWithMultiTimeoutsAsync(CircuitBreakerRetryTest.java:487)
```

**Comment by gsmet** [comment]
Yeah the timeout for failsafe is different so we would need to set both: failsafe.timeout.

Probably something we could add to the common args. It won't hurt anyway.

**Comment by famod** [comment]
I think we should better add it to pom.xml (a base value in build-parent which is overridden in a few places, e.g. in the maven tests), reducing noise in the workflow file.

**Comment by famod** [comment]
Just known flakes, merging...

**Pull request merged**